src/firestore/api-types.ts

```diff
@@ -81,3 +81,30 @@ export interface IndexConfig {
   ancestorField?: string;
   indexes?: Index[];
 }
+
+export interface Location {
+  name: string;
+  labels: any;
+  metadata: any;
+  locationId: string;
+  displayName: string;
+}
+
+export interface DatabaseReq {
+  locationId: string;
+  type: string;
+}
+
+export interface DatabaseResp {
+  name: string;
+  uid: string;
+  createTime: string;
+  updateTime: string;
+  locationId: string;
+  type: string;
+  concurrencyMode: string;
+  appEngineIntegrationMode: string;
+  keyPrefix: string;
+  deleteProtectionState: string;
+  etag: string;
```

Comment:
Forget strongly-typed languages, the future is stringly-typed!

&lt;/joke&gt;

Reply:
Optional - I've been partial to string union types instead of JS enums, as they require a little less type casting when used.  You could consider them instead:

deleteProtectionState: "DELETE_PROTECTION_ENABLED" || "DELETE_PROTECTION_DISABLED"